internal/bminventory/inventory.go

```diff
@@ -5866,3 +5878,55 @@ func isBaremetalBinaryFromAnotherReleaseImageRequired(cpuArchitecture, version s
 		featuresupport.IsFeatureSupported(version,
 			models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING)
 }
+
+// updateMonitoredOperators checks the content of the installer configuration and updates the list
+// of monitored operators accordingly. For example, if the installer configuration uses the
+// capabilities mechanism to disable the console then the console operator is removed from the list
+// of monitored operators.
+func (b *bareMetalInventory) updateMonitoredOperators(tx *gorm.DB, cluster *common.Cluster) error {
+	// Get the complete installer configuration, including the overrides:
+	installConfigData, err := b.installConfigBuilder.GetInstallConfig(cluster, false, "")
+	if err != nil {
+		return err
+	}
+	var installConfig installcfgdata.InstallerConfigBaremetal
+	err = yaml.Unmarshal(installConfigData, &installConfig)
+	if err != nil {
+		return err
+	}
+
+	// Since version 4.11 it is possible to disable the console via the capabilities section of
+	// the installer configuration. The way to do it is to set the base capability set to `None`
+	// and then explicitly list all the enabled capabilities.
+	consoleEnabled := true
+	firstVersion, err := semver.NewVersion("4.11")
+	if err != nil {
+		return err
+	}
+	currentVersion, err := semver.NewVersion(cluster.OpenshiftVersion)
+	if err != nil {
+		return err
+	}
+	if currentVersion.Compare(firstVersion) >= 0 {
+		capabilities := installConfig.Capabilities
+		if capabilities != nil && capabilities.BaselineCapabilitySet == "None" {
+			consoleEnabled = false
+			for _, capability := range capabilities.AdditionalEnabledCapabilities {
+				if capability == "Console" {
+					consoleEnabled = true
+					break
+				}
+			}
+		}
+	}
+
+	// Add or remove the console operator to the list of monitored operators:
+	consoleOperator := operators.OperatorConsole
+	consoleOperator.ClusterID = *cluster.ID
+	if consoleEnabled {
+		err = tx.FirstOrCreate(&consoleOperator).Error
```

Comment:
Is this really required?

Reply:
It isn't strictly required. I will remove it.

Reply:
Actually it is required, because the user may send the request to update the installer confguration multiple times before starting the installation. For example, first to disable the console and then change her mind and enable it again.

Reply:
Hmm, in that case , why do we do it in update cluster logic? Why not to move this part to prepare for installation ?

Reply:
WIll UI will be ok if we remove console? Do they expect it ?

Reply:
@tsorya other operators are added/removed from the `monitored_operators` as soon as they are enabled/disabled in the API/UI. What would be the reason to do it differently for the console?

@nirfarkas can you answer the above question about the UI? I didn't see any effect of enabling/disabling the console, other than the link provided at the end of the installation will not work.

Reply:
As we don't set those in UI, this is internal currently, though i am fine with the way if we plan in future to add it to UI